### PR TITLE
Disable DD's log_injection in development

### PR DESCRIPTION
**Pull request by maestromac**
## What type of PR is this? (check all applicable)
- [x] DX

## Description
I did not realize that DataDog's log_injection was applied to all, making server log a pain to use in development. This will hopefully improve DX.

## Added tests?
- [x] no, because they aren't needed

## Added to documentation?
- [x] no documentation needed

## [optional] Are there any post deployment tasks we need to perform?
n/a

### CODE REVIEW

File: config/initializers/datadog.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@
   c.env = Rails.env
   c.tracing.enabled = ENV["DD_API_KEY"].present?
   c.tracing.partial_flush.enabled = true
+  c.tracing.log_injection = !Rails.env.development?
```
Comment (klardotsh) [inline config/initializers/datadog.rb:5]: Nit: should this also check that we're not in unit tests?

**Review by klardotsh — APPROVED** [review]

File: config/initializers/datadog.rb (hunk 1 above)
Comment (maestromac) [reply config/initializers/datadog.rb:5]: yeah actually I think the only environment we want this enabled is prod

Comment (maestromac) [reply config/initializers/datadog.rb:5]:
```suggestion
  c.tracing.log_injection = Rails.env.production?
```

**Review by jaw6 — APPROVED** [review]
Works well, looks great, removes another item from my local perma-stash :+1: 🙌 :smile: :tada:

**Pull request merged**